### PR TITLE
Image cropper: remove clientWidth prop with useResizeObserver

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Removes `useClientWidth` in favour of using `useResizeObserver`.

## Why?

This has the advantage of (1) being easier to use by consumers and (2) not calculating the clientWidth on block mount, which is expensive and causes a layout calculation.

## How?

We can use `useResizeObserver` inside the cropper component.

## Testing Instructions

⚠️ **Make sure to `npm install` after the change in #60581, otherwise this fix won't work as expected.**

Create an image block. Click on the cropper button. The cropper area should be correct (images should be same width as before you pressed the cropper button).
Now switch between align none and align full. The cropper area should adjust correctly.

You can also test if the cropper works for the Site Logo block.

### CODE REVIEW

File: packages/block-library/src/site-logo/edit.js [hunk 1]
```diff
@@ -154,14 +146,7 @@ const SiteLogo = ( {
...
 	}
 
-	let imageWidthWithinContainer;
-
-	if ( clientWidth && naturalWidth && naturalHeight ) {
-		const exceedMaxWidth = naturalWidth > clientWidth;
-		imageWidthWithinContainer = exceedMaxWidth ? clientWidth : naturalWidth;
-	}
-
-	if ( ! isResizable || ! imageWidthWithinContainer ) {
```
Comment (ellatrix) [inline packages/block-library/src/site-logo/edit.js:164]: I'm not sure I understood why this condition was done this way. `imageWidthWithinContainer` returns either `clientWidth` or `naturalWidth`. Which will always be a number unless something went wrong with the image loading. So a check for `! naturalWidth || ! naturalHeight` should be enough here, to make sure the image loaded.

Comment (ellatrix) [reply packages/block-library/src/site-logo/edit.js:164]: This goes all the way back to #18811, cc @scruffian

**Review by youknowriad — APPROVED** [review]
Nice simplification, I think the useResizeObserver hook didn't exist yet when we had this clientWidth thing in place.

File: packages/block-library/src/site-logo/edit.js (hunk 1 above)
Comment (scruffian) [reply packages/block-library/src/site-logo/edit.js:164]: I'm sorry I don't remember!

Comment (ellatrix) [reply packages/block-library/src/site-logo/edit.js:164]: It's fine, I think we're good here

**Pull request merged**